File: C/section08/S08E06.c
#include <stdio.h>

int main() { 
    int optionCode = 1;
    float numbers[5];

    while (optionCode != 0) {
        printf ("\nPlease enter the option code: \n2 - Show the vector in reversed as entered order.\n1 - Show the vector in the entered order.\n0 - Quit.\n");
        scanf("%d", &optionCode);

        if (optionCode == 2 || optionCode == 1) {
            for (int count = 0; count < 5; count++) {
            printf("Please enter a number: ");
            scanf("%f", &numbers[count]);
            }

            if (optionCode == 1) {
                printf("Showing the vector in the entered order: \n");
                for (int count = 0; count < 5; count++) {
                    printf("%.2f\n", numbers[count]);
                }
            } else if (optionCode == 2) {
                printf("Showing the vector in reversed as entered order: \n");
                for (int count = 4; count >= 0; count--) {
                    printf("%.2f\n", numbers[count]);
                }
            }
            
        } else if (optionCode == 0) {
            printf("You choose to quit.");
        } else {
            printf("Please enter a number between 0 and 2.\n");
        }
    }

    return 0;
}

File: Python/section08/S08e06.py
vector = []
optionCode = -1

while optionCode != 0:
    optionCode = int(input("\nPlease choose an option:\n1 - Show the numbers in direct order.\n2 - Show the numbers in reverse order.\n0 - Quit.\n"))

    if optionCode < 0 or optionCode > 2:
        while optionCode < 0 or optionCode > 2:
            optionCode = int(input("No such code, please try again:\n1 - Show the numbers in direct order.\n2 - Show the numbers in reverse order.\n0 - Quit.\n"))
    elif optionCode == 1 or optionCode == 2:
        for index in range(0,5):
            vector.append(float(input("Please enter a number for {0}/5: ".format(index+1))))
        
        if optionCode == 2:
            vector.reverse()
            print("\nShowing in inverse order: ")
            for index in range(0,5):
                print(vector[index])
        elif optionCode == 1:
            print("\nShowing in direct order: ")
            for index in range(0,5):
                print(vector[index])
    else:
            print("You chose to quit.")
        